### DISCUSSION TITLE
Allow to drop metrics by label sub value

### DISCUSSION THREAD

**Discussion by asaf1602** [prometheus/prometheus#11824] [Q&A]
### Proposal

We have in prometheus a huge number of instances of the metrics node_systemd_unit_state (count >10k) arrived from various of VMs by node exporter.
node_systemd_unit_state{ host="XXXXXXX", instance="xx.xx1.xx2.xx:yyyy", job="node", name="polkit.service", ...}
node_systemd_unit_state{ host="XXXXXXX", instance="xx.xx1.xx2.xx:yyyy", job="node", name="systemd-logind.service", ...} 0
node_systemd_unit_state{ host="XXXXXXX", instance="xx.xx1.xx2.xx:yyyy", job="node", name="tuned.service", ...}
node_systemd_unit_state{ host="XXXXXXX", instance="xx.xx1.xx2.xx:yyyy", job="node", name="auditd.service", ...}
node_systemd_unit_state{ host="XXXXXXX", instance="xx.xx1.xx2.xx:yyyy", job="node", name="chronyd.service", ...}
node_systemd_unit_state{ host="XXXXXXX", instance="xx.xx1.xx2.xx:yyyy", job="node", name="gssproxy.service", ...}
node_systemd_unit_state{ host="XXXXXXX", instance="xx.xx1.xx2.xx:yyyy", job="node", name="network.service", c
node_systemd_unit_state{ host="XXXXXXX", instance="xx.xx1.xx2.xx:yyyy", job="node", name="nfs-idmapd.service", ...}
node_systemd_unit_state{ host="XXXXXXX", instance="xx.xx1.xx2.xx:yyyy", job="node", name="postfix.service", ...}
node_systemd_unit_state{ host="XXXXXXX", instance="xx.xx1.xx2.xx:yyyy", job="node", name="rc-local.service", ...}
node_systemd_unit_state{ host="XXXXXXX", instance="xx.xx1.xx2.xx:yyyy", job="node", name="MY-Service-1", ...}
node_systemd_unit_state{ host="XXXXXXX", instance="xx.xx1.xx2.xx:yyyy", job="node", name="MY-Service-2", ...}
...

total different names in label "name" are 158.

I want to drop all those metrics where name!="MY-Service.*"
I try to use below options of regex but non of them isnt work for me.

metric_relabel_configs:
- source_labels: [__name__, name]
  regex: 'node_systemd_unit_state;"(.*)target|atd(.*)|auditd(.*)|authz(.*)|autoz(.*)|chronyd(.*)|cloud-(.*)|cpupower(.*)|crond(.*)|cups(.*)|dbus(.*)|dracut-(.*)|emergency(.*)|filebeat(.*)|getty(.*)|gssproxy(.*)|initrd-(.*)|irqbalance(.*)|kdump(.*)|kmod(.*)|microcode(.*)|network(.*)|nfs(.*)|polkit(.*)|postfix(.*)|rc-local(.*)|rescue(.*)|rhel(.*)|rpc(.*)|rsyslog(.*)|*targeted|serial(.*)|sshd(.*)|(.*)socket|sysstat(.*)|tuned(.*)|zing-(.*)"'
  action: drop

- source_labels: [__name__, name]
  regex: 'node_systemd_unit_state;".*target|atd.*|auditd.*|authz.*|autoz.*|chronyd.*|cloud-.*|cpupower.*|crond.*|cups.*|dbus.*|dracut-.*|emergency.*|filebeat.*|getty.*|gssproxy.*|initrd-.*|irqbalance.*|kdump.*|kmod.*|microcode.*|network.*|nfs.*|polkit.*|postfix.*|rc-local.*|rescue.*|rhel.*|rpc.*|rsyslog.*|*targeted|serial.*|sshd.*|.*socket|sysstat.*|tuned.*|zing-.*"'
  action: drop

- source_labels: [__name__]
  regex: 'node_systemd_unit_state{name=~".*target|atd.*|auditd.*|authz.*|autoz.*|chronyd.*|cloud-.*|cpupower.*|crond.*|cups.*|dbus.*|dracut-.*|emergency.*|filebeat.*|getty.*|gssproxy.*|initrd-.*|irqbalance.*|kdump.*|kmod.*|microcode.*|network.*|nfs.*|polkit.*|postfix.*|rc-local.*|rescue.*|rhel.*|rpc.*|rsyslog.*|*targeted|serial.*|sshd.*|.*socket|sysstat.*|tuned.*|zing-.*"'
  action: drop

- source_labels: [__name__]
  regex: 'node_systemd_unit_state{name!~"MY-Service.*"'
  action: drop

**Comment by roidelapluie** [comment]
```
metric_relabel_configs:
- source_labels: [__name__]
  regex: 'node_systemd_unit_state'
  target_label: __tmp_keep
  replacement: 'no'
- source_labels: [__name__,name]
  regex: 'node_systemd_unit_state;MY-service.*'
  target_label: __tmp_keep
  replacement: 'yes'
- source_labels: [__tmp_keep]
  regex: 'no'
  action: drop
- regex: __tmp_keep
  action: labeldrop
```

**Reply by skywalkr** [reply]
I don't want to share what my solution looked like before seeing this answer, but it was ugly. This can also be applied to windows_service metrics from the windows_exporter.
